space/internal/spaceprocess/joiner/statuschanger.go

```diff
@@ -0,0 +1,39 @@
+package joiner
+
+import (
+	"context"
+
+	"github.com/anyproto/any-sync/app"
+
+	"github.com/anyproto/anytype-heart/space/internal/components/spacestatus"
+	"github.com/anyproto/anytype-heart/space/spaceinfo"
+)
+
+const statusChangerCName = "joiner.statuschanger"
+
+type statusChanger struct {
+	status spacestatus.SpaceStatus
+}
+
+func newStatusChanger() *statusChanger {
+	return &statusChanger{}
+}
+
+func (s *statusChanger) Init(a *app.App) (err error) {
+	s.status = a.MustComponent(spacestatus.CName).(spacestatus.SpaceStatus)
+	return nil
+}
+
+func (s *statusChanger) Name() (name string) {
+	return statusChangerCName
+}
+
+func (s *statusChanger) Run(ctx context.Context) (err error) {
+	s.status.Lock()
```

Comment:
why create separate component only to do one thing that could be done in?

```
func (i *joiner) Start(ctx context.Context) error {
	return i.app.Start(ctx)
}
```

Reply:
No this couldn't be done like this, because this status thing might not be initialised at that point (it sets techSpace on init). I could've done it better by presumably making status a component with empty init and initialising it via constructor essentially. But yeah now it works like this :-)

Reply:
So you are right that this is weird, I will move techspace to a status constructor init at some point, just need to check that this doesn't break anything